File: app/src/main/java/com/dertyp7214/appstore/fragments/TabFragment.java
/*
 * Copyright (c) 2018.
 * Created by Josua Lengwenath
 */

package com.dertyp7214.appstore.fragments;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.view.View;
import android.view.ViewGroup;

import com.dertyp7214.appstore.Config;

import java.io.IOException;
import java.net.InetAddress;
import java.net.InetSocketAddress;
import java.net.Socket;
import java.net.SocketAddress;
import java.net.URL;

public class TabFragment extends Fragment {

    public String getName(Context context){
        return "";
    }

    public boolean serverOnline(){
        try {
            URL url = new URL(Config.API_URL);
            SocketAddress sockaddr = new InetSocketAddress(InetAddress.getByName(url.getHost()), 80);
            Socket sock = new Socket();
            int timeoutMs = 2000;
            sock.connect(sockaddr, timeoutMs);
            return true;
        } catch(IOException ignored) {
        }
        return false;
    }

    public boolean haveConnection(){
        try {
            URL url = new URL("http://www.google.de");
            SocketAddress sockaddr = new InetSocketAddress(InetAddress.getByName(url.getHost()), 80);
            Socket sock = new Socket();
            int timeoutMs = 2000;
            sock.connect(sockaddr, timeoutMs);
            return true;
        } catch(IOException ignored) {
        }
        return false;
    }

    public int getNavigationBarHeight() {
        int result = 0;
        int resourceId = getResources().getIdentifier("navigation_bar_height", "dimen", "android");
        if (resourceId > 0) {
            result = getResources().getDimensionPixelSize(resourceId);
        }
        return result;
    }

    public void setMargins (View v, int l, int t, int r, int b) {
        if (v.getLayoutParams() instanceof ViewGroup.MarginLayoutParams) {
            ViewGroup.MarginLayoutParams p = (ViewGroup.MarginLayoutParams) v.getLayoutParams();
            if(l == -1) l = p.leftMargin;
            if(t == -1) t = p.topMargin;
            if(r == -1) r = p.rightMargin;
            if(b == -1) b = p.bottomMargin;
            p.setMargins(l, t, r, b);
            v.requestLayout();
        }
    }
}

File: app/src/main/java/com/dertyp7214/appstore/fragments/TabFragment.kt
/*
 * Copyright (c) 2018.
 * Created by Josua Lengwenath
 */

package com.dertyp7214.appstore.fragments

import android.content.Context
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.dertyp7214.appstore.Config
import java.io.IOException
import java.net.InetAddress
import java.net.InetSocketAddress
import java.net.Socket
import java.net.URL

@Suppress("NAME_SHADOWING")
open class TabFragment : Fragment() {

    val navigationBarHeight: Int
        get() {
            var result = 0
            val resourceId = resources.getIdentifier("navigation_bar_height", "dimen", "android")
            if (resourceId > 0) {
                result = resources.getDimensionPixelSize(resourceId)
            }
            return result
        }

    open fun getName(context: Context): String {
        return ""
    }

    fun serverOnline(): Boolean {
        return try {
            val url = URL(Config.API_URL)
            val sockaddr = InetSocketAddress(InetAddress.getByName(url.host), 80)
            val sock = Socket()
            val timeoutMs = 2000
            sock.connect(sockaddr, timeoutMs)
            true
        } catch (ignored: IOException) {
            false
        }
    }

    fun haveConnection(): Boolean {
        return try {
            val url = URL("http://www.google.de")
            val sockaddr = InetSocketAddress(InetAddress.getByName(url.host), 80)
            val sock = Socket()
            val timeoutMs = 2000
            sock.connect(sockaddr, timeoutMs)
            true
        } catch (ignored: IOException) {
            false
        }
    }

    fun setMargins(v: View, l: Int, t: Int, r: Int, b: Int) {
        if (v.layoutParams is ViewGroup.MarginLayoutParams) {
            val p = v.layoutParams as ViewGroup.MarginLayoutParams
            val l = if (l == -1) p.leftMargin else l
            val t = if (t == -1) p.topMargin else t
            val r = if (r == -1) p.rightMargin else r
            val b = if (b == -1) p.bottomMargin else b
            p.setMargins(l, t, r, b)
            v.requestLayout()
        }
    }
}
